### ISSUE TITLE
Stuck at 100% CPU usage

### ISSUE THREAD

**Issue by af7567** [imerzan/ArmoryBot#2]
Running on Linux x86 with .net core 3.1.13. Using the precompiled release and compiling it myself I have the same problem.
Everything seems to be working fine, but it is using 100% CPU all the time. I think it's something to do with this line:

https://github.com/imerzan/ArmoryBot/blob/96b63e6ecdae583703d978e3f1d20e159c6a677c/src/Program.cs#L33

I'm not really sure what the Task.Delay does here. I thought it was just meant to stop Main from exiting, so I replaced it with Thread.Sleep(1000); for now to stop the high CPU usage. The bot still works, but it's giving an unhandled exception sometimes.

**Comment by imerzan** [comment]
Yeah, I was looking at that earlier and realized Task.Delay isn't doing anything there since it's not awaited. 

Any idea where the unhandled exception is being thrown? I can take a look and get this fixed later tonight.

**Comment by af7567** [comment]
Thanks, I have built the issue-#2  branch and the high CPU usage is gone now.

As for the uncaught exception message I saw it was something to do with Discord.Net but i'm afraid I don't have the errors now because I wasn't logging anything to a file. I just remember an object was null at line 39 of ArmoryBot.cs
```
  if (msg.Author.IsBot) return; // Bot Check
```
But I didn't see this before I added my Thread.Sleep() so I thought it was maybe not reading from Discord correctly because of what I changed. As you said it might have just been the Discord reconnects - the armory lookup requests still seemed to work OK.

**Comment by imerzan** [comment]
I think I found the problem in the MsgReceived handler, and fixed it with commit 5787d48

https://github.com/imerzan/ArmoryBot/blob/5787d48f6471759c68addeb0f1bb40bebe69177c/src/ArmoryBot.cs#L37-L38

This should prevent any NullReferences from slipping through :)

**Comment by imerzan** [comment]
New pre-compiled release available
https://github.com/imerzan/ArmoryBot/releases/tag/v1.0.902.201

**Comment by af7567** [comment]
All seems to be working now with the new precompiled linux-x86 release :)
Still get the occasional disconnects like you said, but ArmoryBot is successfully reconnecting when that happens.
```
3/12/2021 4:19:31 PM: 16:19:31 Gateway     System.Exception: WebSocket connection was closed
 ---> System.Net.WebSockets.WebSocketException (0x80004005): The remote party closed the WebSocket connection without completing the close handshake.
 ---> System.Net.WebSockets.WebSocketException (0x80004005): The remote party closed the WebSocket connection without completing the close handshake.
   at System.Net.WebSockets.ManagedWebSocket.ThrowIfEOFUnexpected(Boolean throwOnPrematureClosure)
   at System.Net.WebSockets.ManagedWebSocket.EnsureBufferContainsAsync(Int32 minimumRequiredBytes, CancellationToken cancellationToken, Boolean throwOnPrematureClosure)
   at System.Net.WebSockets.ManagedWebSocket.ReceiveAsyncPrivate[TWebSocketReceiveResultGetter,TWebSocketReceiveResult](Memory`1 payloadBuffer, CancellationToken cancellationToken, TWebSocketReceiveResultGetter resultGetter)
   at System.Net.WebSockets.ManagedWebSocket.ReceiveAsyncPrivate[TWebSocketReceiveResultGetter,TWebSocketReceiveResult](Memory`1 payloadBuffer, CancellationToken cancellationToken, TWebSocketReceiveResultGetter resultGetter)
   at Discord.Net.WebSockets.DefaultWebSocketClient.RunAsync(CancellationToken cancelToken)
   --- End of inner exception stack trace ---
   at Discord.ConnectionManager.<>c__DisplayClass29_0.<<StartAsync>b__0>d.MoveNext()
3/12/2021 4:19:31 PM: 16:19:31 Gateway     Disconnecting
3/12/2021 4:19:31 PM: 16:19:31 Gateway     Disconnected
3/12/2021 4:19:32 PM: 16:19:32 Gateway     Connecting
3/12/2021 4:19:33 PM: 16:19:33 Gateway     Resumed previous session
3/12/2021 4:19:33 PM: 16:19:33 Gateway     Connected
```

**Comment by imerzan** [comment]
@af7567 Released a pre-release version for 9.1 that made significant improvements to the bot client. May be more stable than the last release since this issue. Hope this helps!
release: https://github.com/imerzan/ArmoryBot/releases/tag/v0.1.91.0
readme: https://github.com/imerzan/ArmoryBot/tree/9.1-release
